web/packages/teleport/src/Bots/Add/GitHubActionsK8s/KubernetesLabelsSelect.test.tsx

```diff
@@ -0,0 +1,400 @@
+
+    expect(
+      within(modal).getByRole('heading', { name: 'Selected Labels (3)' })
+    ).toBeInTheDocument();
+  });
+
+  test('add label', async () => {
+    withListUnifiedResourcesSuccess();
+
+    const { user } = renderComponent();
+
+    const edit = screen.getByRole('button', { name: 'edit' });
+    await user.click(edit);
+
+    const modal = screen.getByTestId('Modal');
+
+    expect(
+      within(modal).getByRole('heading', { name: 'Selected Labels (0)' })
+    ).toBeInTheDocument();
+
+    const item = screen.getByTestId('cluster-item-kube-lon-dev-01.example.com');
+    const addLabel = within(item).getByTestId('label-action-env: dev-add');
+    await user.click(addLabel);
+
+    expect(
+      within(modal).getByRole('heading', { name: 'Selected Labels (1)' })
+    ).toBeInTheDocument();
+
+    const selectedSection = within(modal)
+      .getByRole('heading', { name: 'Selected Labels (1)' })
+      .closest('div');
+    expect(within(selectedSection!).getByText('env: dev')).toBeInTheDocument();
+  });
+
+  test('remove label', async () => {
+    withListUnifiedResourcesSuccess();
+
+    const { user } = renderComponent({
+      props: {
+        selected: [
+          {
+            name: 'env',
+            values: ['dev'],
+          },
+        ],
+      },
+    });
+
+    const edit = screen.getByRole('button', { name: 'edit' });
+    await user.click(edit);
+
+    const modal = screen.getByTestId('Modal');
+
+    expect(
+      within(modal).getByRole('heading', { name: 'Selected Labels (1)' })
+    ).toBeInTheDocument();
+
+    const selectedSection = within(modal)
+      .getByRole('heading', { name: 'Selected Labels (1)' })
+      .closest('div');
+    const removeLabel = within(selectedSection!).getByTestId(
+      'label-action-env: dev-remove'
+    );
+    await user.click(removeLabel);
+
+    const item = screen.getByTestId('cluster-item-kube-lon-dev-01.example.com');
+    expect(
+      within(item).getByTestId('label-action-env: dev-add')
+    ).toBeInTheDocument();
+
+    expect(
+      within(modal).getByRole('heading', { name: 'Selected Labels (0)' })
+    ).toBeInTheDocument();
+    expect(
+      within(selectedSection!).queryByText('env: dev')
+    ).not.toBeInTheDocument();
+  });
+
+  test('manual entry', async () => {
+    withListUnifiedResourcesSuccess();
+
+    const { user } = renderComponent();
+
+    const edit = screen.getByRole('button', { name: 'edit' });
+    await user.click(edit);
+
+    const modal = screen.getByTestId('Modal');
+
+    const selectedSection = within(modal)
+      .getByRole('heading', { name: 'Selected Labels (0)' })
+      .closest('div');
+
+    const input = within(selectedSection!).getByPlaceholderText('name: value');
+    await user.type(input, 'foo:bar{enter}');
+    await user.type(input, 'env:*{enter}');
+    // Extra square brackets are required to escape the initial pair
+    await user.type(input, 'region:^eu-(west|east)-[[0-9]]+$');
```

Comment:
today i learned that you can supply regex to role resource labels (wildcard i knew about but not regex)